main.py

```diff
@@ -115,6 +115,17 @@ def handle_payload(jsonData, is_retry_payload=False):
                 and not is_retry_payload
             ):
                 handle_user_custom_field_data(jsonData)
+
+            # Handle groups and fields
+            if (
+                "fields"
+                or "groups" in jsonData["contact"]
+                and jsonData["contact"]["fields"]
+                or jsonData["contact"]["groups"] is not None
+                and jsonData.get("flow_category", None) == "dry_flow"
+                and not is_retry_payload
+            ):
```

Comment:
This will fail when 
- groups in jsonData["contact"] 
- fields not in  jsonData["contact"] 
In this case jsonData["contact"]["fields"] will throw error that `fields` key does not exist

Reply:
Got it.